subjects/20-imperative-to-declarative/exercise.js

```diff
@@ -12,10 +12,35 @@ import $ from 'jquery'
 import 'bootstrap-webpack'
 
 const Modal = React.createClass({
+  getDefaultProps() {
+    return {
+      isOpen: false
+    }
+  },
+
+  componentDidMount() {
+    this.showOrHideModal()
+  },
+
+  componentDidUpdate() {
+    this.showOrHideModal()
+  },
+
+  closeModal() {
+    this.props.onModalClosed()
+  },
+
+  showOrHideModal() {
+    if (this.props.isOpen) {
+      $(findDOMNode(this)).modal('show')
+    } else {
+      $(findDOMNode(this)).modal('hide')
+    }
+  },
 
   render() {
     return (
-      <div className="modal fade">
+      <div className="modal fade" data-backdrop="static">
```

Comment:
Avoiding listening to that `bs.modal.close` event. User experience is worse. Don't do this!